libarchive/archive_read_disk_entry_from_file.c

```diff
@@ -437,20 +453,27 @@ setup_acls(struct archive_read_disk *a,
 #endif
 	else
 		acl = acl_get_file(accpath, ACL_TYPE_NFS4);
+
 #if HAVE_ACL_IS_TRIVIAL_NP
-	/* Ignore "trivial" ACLs that just mirror the file mode. */
-	acl_is_trivial_np(acl, &r);
-	if (r) {
-		acl_free(acl);
-		acl = NULL;
+	if (acl != NULL && acl_is_trivial_np(acl, &r) == 0) {
```

Comment:
We definitely do not want to check if a NULL ACL is trivial and we do not want to deal with unintialized r if acl_is_trivial_np() fails.